pandas/core/internals/managers.py

```diff
@@ -827,8 +827,9 @@ def _slice_take_blocks_ax0(
                     # A non-consolidatable block, it's easy, because there's
                     # only one item and each mgr loc is a copy of that single
                     # item.
+                    deep = not (only_slice or using_copy_on_write())
                     for mgr_loc in mgr_locs:
-                        newblk = blk.copy(deep=not only_slice)
+                        newblk = blk.copy(deep=deep)
```

Comment:
Could this have caused ``test_reindex_copies_ea`` to fail? Not sure but can't see how this test could have been impacted otherwise

Reply:
Ah, yes, that's the test that I added in the precursor PR. With CoW we now don't copy (the test was added in the assumption we always copy with copy=True, while now we don't copy in case of CoW)

Reply:
But, for most other methods I think we are actually still honoring the `copy=True` keyword .. Should we do that here as well for reindex with EAs?

(although we should also stop honoring the `copy=True` keyword at some point, when CoW is enabled, xref https://github.com/pandas-dev/pandas/issues/50535)

Reply:
Yeah I would prefer not honouring copy anymore for CoW. I'll start tracking the cases down where we are still using copy.